### PR TITLE
Add Relocators

### PR DESCRIPTION
**Pull request by BluePsychoRanger**
Adds an item that allows easy transportation of gm4 machine blocks such as custom crafters and compressors

https://docs.google.com/document/d/1KvPG56kB-GiVh2uWMlXxDjXEXq125xOPGoIkXqW4hpA/edit?usp=sharing

### CODE REVIEW

**Comment by sparks0** [comment]
@BluePsychoRanger Just a reminder there are changes requested.

**Comment by BluePsychoRanger** [comment]
I've just done some testing, and even when I watched the stream replay for the first time, it seemed kinda fishy what the bug was. The bug is not an item voiding bug; if you place down all the compressors you will get back all the items used to create the two machines, but they won't spawn in properly. It is a bug, and I believe I know how to fix it, but it seems that your testing wasn't really correct and you just got the wrong items for what the machine was supposed to be. 
I. e. the compressor you got from picking it up with the relocator had the custom crafter armor stand data, which made it seem that the compressor items were voided when placing it back down, but in reality, picking up the compressor with the relocator would drop the items of the compressor first, then store custom crafter data into the relocator. 
I've tested this in survival and no items were voided by reproducing those steps and trying to reproduce the results

**Comment by BluePsychoRanger** [comment]
TL;DR 
I know what the bug is, but it's not an item voiding bug. It's an incorrect storing data bug.

**Comment by BluePsychoRanger** [comment]
@sparks0 this module has been fixed and is ready for testing

File: gm4_relocators/data/relocators/functions/recipe_check.mcfunction [hunk 1]
```diff
@@ -0,0 +1 @@
+execute if score @s gm4_slot_count matches 9 if score @s gm4_stack_size matches ..64 if block ~ ~ ~ dropper{Items:[{Slot:0b,id:"minecraft:diamond"},{Slot:1b,id:"minecraft:popped_chorus_fruit"},{Slot:2b,id:"minecraft:diamond"},{Slot:3b,id:"minecraft:popped_chorus_fruit"},{Slot:4b,id:"minecraft:nether_star"},{Slot:5b,id:"minecraft:popped_chorus_fruit"},{Slot:6b,id:"minecraft:diamond"},{Slot:7b,id:"minecraft:popped_chorus_fruit"},{Slot:8b,id:"minecraft:diamond"}]} run data merge block ~ ~ ~ {Items:[{Slot:8,id:"player_head",Count:1b,tag:{CustomModelData:26,gm4_custom_crafters:{multiplier:1},gm4_relocator:1b,display:{Name:'{"translate":"%1$s%3427655$s","with":["Relocator",{"translate":"item.gm4.relocator"}],"italic":false}'},SkullOwner:{Id:"2c0fa9da-b8e7-4dfe-81eb-7885c62d6a92",Properties:{textures:[{Value:"eyJ0aW1lc3RhbXAiOjE1NjM4MTY4ODE5NTcsInByb2ZpbGVJZCI6IjdkYTJhYjNhOTNjYTQ4ZWU4MzA0OGFmYzNiODBlNjhlIiwicHJvZmlsZU5hbWUiOiJHb2xkYXBmZWwiLCJzaWduYXR1cmVSZXF1aXJlZCI6dHJ1ZSwidGV4dHVyZXMiOnsiU0tJTiI6eyJ1cmwiOiJodHRwOi8vdGV4dHVyZXMubWluZWNyYWZ0Lm5ldC90ZXh0dXJlLzMwM2Q3YTkyNTIxZDZmZTU1M2ZiNGYyNzIxOTI2Yzk4YTVlMDdlMDhmZjdiNzg4OTg1OTc3MGM0Yjg0NmJlZDYifX19"}]}}}}]}
```
Comment (BluePsychoRanger) [inline gm4_relocators/data/relocators/functions/recipe_check.mcfunction]:
```suggestion
execute if score @s gm4_slot_count matches 9 if score @s gm4_stack_size matches ..64 if block ~ ~ ~ dropper{Items:[{Slot:0b,id:"minecraft:diamond"},{Slot:1b,id:"minecraft:popped_chorus_fruit"},{Slot:2b,id:"minecraft:diamond"},{Slot:3b,id:"minecraft:popped_chorus_fruit"},{Slot:4b,id:"minecraft:enchanted_book",tag:{StoredEnchantments:[{id:"minecraft:silk_touch"}]}},{Slot:5b,id:"minecraft:popped_chorus_fruit"},{Slot:6b,id:"minecraft:diamond"},{Slot:7b,id:"minecraft:popped_chorus_fruit"},{Slot:8b,id:"minecraft:diamond"}]} run data merge block ~ ~ ~ {Items:[{Slot:8,id:"player_head",Count:1b,tag:{CustomModelData:26,gm4_custom_crafters:{multiplier:1},gm4_relocator:1b,display:{Name:'{"translate":"%1$s%3427655$s","with":["Relocator",{"translate":"item.gm4.relocator"}],"italic":false}'},SkullOwner:{Id:"2c0fa9da-b8e7-4dfe-81eb-7885c62d6a92",Properties:{textures:[{Value:"eyJ0aW1lc3RhbXAiOjE1NjM4MTY4ODE5NTcsInByb2ZpbGVJZCI6IjdkYTJhYjNhOTNjYTQ4ZWU4MzA0OGFmYzNiODBlNjhlIiwicHJvZmlsZU5hbWUiOiJHb2xkYXBmZWwiLCJzaWduYXR1cmVSZXF1aXJlZCI6dHJ1ZSwidGV4dHVyZXMiOnsiU0tJTiI6eyJ1cmwiOiJodHRwOi8vdGV4dHVyZXMubWluZWNyYWZ0Lm5ldC90ZXh0dXJlLzMwM2Q3YTkyNTIxZDZmZTU1M2ZiNGYyNzIxOTI2Yzk4YTVlMDdlMDhmZjdiNzg4OTg1OTc3MGM0Yjg0NmJlZDYifX19"}]}}}}]}
```

Comment (sparks0) [reply gm4_relocators/data/relocators/functions/recipe_check.mcfunction]: I can't commit the suggestion XD
